pkg/solana/chainreader/event_read_binding.go

```diff
@@ -286,38 +292,41 @@ func (b *eventReadBinding) normalizeParams(value any, itemType string) (any, err
 	return offChain, nil
 }
 
-func (b *eventReadBinding) extractFilterSubkeys(offChainParams any) (query.Expression, error) {
+func (b *eventReadBinding) extractFilterSubkeys(offChainParams any) ([]query.Expression, error) {
 	var expressions []query.Expression
 
 	for offChainKey, idx := range b.indexedSubKeys.lookup {
 		itemType := codec.WrapItemType(true, b.namespace, b.genericName+"."+offChainKey)
 
-		onChainValue, err := b.modifier.TransformToOnChain(offChainParams, itemType)
+		fieldVal, err := valueForPath(reflect.ValueOf(offChainParams), offChainKey)
```

Comment:
Oh... I thought an offchain key was just supposed to be some generic key name, is it actually a path sometimes?

Reply:
Yes, it is a path to a value in a struct.